core/engine_test.go

```diff
@@ -187,10 +187,17 @@ func TestEngineRun(t *testing.T) {
 
 		runner := &minirunner.MiniRunner{
 			Fn: func(ctx context.Context, _ *lib.State, out chan<- metrics.SampleContainer) error {
-				metrics.PushIfNotDone(ctx, out, metrics.Sample{Metric: testMetric, Time: time.Now(), Value: 1})
+				metrics.PushIfNotDone(ctx, out, metrics.Sample{
+					TimeSeries: metrics.TimeSeries{Metric: testMetric, Tags: piState.Registry.RootTagSet()},
```

Comment:
Given how much this need to be written ... maybera "constructor"?

Reply:
I thought about it, but it will become one of these constructors with 5 different parameters, where a struct initialization is just more readable 99% of the time. Moreover, when I add `metadata map[string]string`, which might be `nil`, I will need to add it as another parameter and edit all of the tests to add `nil` everywhere once again... So I decided to stick with this.

Reply:
Maybe even for the whole sample as I see that we in some places have forgotten Time or Value in the past :facepalm:

Reply:
this is what I meant - a constructor for the whole sample will be super cumbersome :disappointed: especially with the upcoming `metadata`

Reply:
My original comment was about only `TimeSeries` 

I guess you won't be putting the `metadata` there so it might still be nicer :thinking: